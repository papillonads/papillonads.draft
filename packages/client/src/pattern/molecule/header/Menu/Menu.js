import React from 'react'
import { string } from 'prop-types'
import cn from 'classnames'
import styles from './Menu.scss'

const Menu = ({ className }) => {
  const menuClassList = cn(className, styles.menu)

  return <div className={menuClassList}>Menu</div>
}

Menu.defaultProps = {
  className: '',
}

Menu.propTypes = {
  className: string,
}.isRequired

export default Menu
