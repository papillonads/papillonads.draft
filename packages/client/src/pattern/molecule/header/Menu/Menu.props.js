import { string } from 'prop-types'

export const menuDefaultProps = {
  className: '',
}

export const menuPropTypes = {
  className: string,
}
