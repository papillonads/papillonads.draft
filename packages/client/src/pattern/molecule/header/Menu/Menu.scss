/* stylelint-disable font-family-no-missing-generic-family-keyword */

@import '../../../../sass/mixin.scss';

._ {
  display: flex;
  justify-content: space-around;

  @media (min-width: 992px) {
    padding: 0;
  }

  &Messages,
  &Notifications {
    @include headerNavigationBarMenuLabelBadge;

    &Icon {
      padding: 0;

      @media (min-width: 992px) {
        padding: 0 12px 0 0;
      }
    }
  }

  &PlaceAd {
    @include headerNavigationBarMenuButtonBadge;

    &Icon {
      transform: scaleX(-1);
      filter: FlipH;
      color: #fff;
      padding: 0;

      @media (min-width: 992px) {
        padding: 0 8px;
      }
    }

    &Button {
      font-family: BreeSerif;
      font-size: 14px;
      align-self: center;
    }
  }

  &MessagesLabel,
  &NotificationsLabel,
  &PlaceAdButton {
    display: none;

    @media (min-width: 992px) {
      display: block;
    }
  }
}

/* stylelint-enable font-family-no-missing-generic-family-keyword */
