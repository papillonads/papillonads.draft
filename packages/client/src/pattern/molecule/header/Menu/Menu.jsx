import React from 'react'
import cn from 'classnames'
import cs from './Menu.scss'
import Badge from '../Badge'
import { SvgIcon } from '../../../atom/header/Icon'
import Label from '../../../atom/header/Label'
import Dropdown from '..//Dropdown'
import Button from '../../../atom/header/Button'
import { menuDefaultProps, menuPropTypes } from './Menu.props'

const Menu = ({ className }) => {
  const dropdownItems = [
    { text: 'Mijn Advertenties' },
    { text: 'Mijn Favorieten' },
    { text: 'Mijn Zoekopdrachten' },
    { text: 'Mijn Verkopers' },
    { text: 'Mijn Profiel' },
    { text: 'Mijn Ervaringen' },
    { text: 'Uitloggen' },
  ]

  return (
    <div className={cn(className, cs._menu)}>
      <Badge className={cs.messages}>
        <SvgIcon className={cs.messagesIcon} fill="grey" name="bubbles4" width={15} height={15} />
        <Label className={cs.messagesLabel} text="Berichten" />
      </Badge>
      <Badge className={cs.notifications}>
        <SvgIcon className={cs.notificationsIcon} fill="grey" name="bell" width={15} height={15} />
        <Label className={cs.notificationsLabel} text="Meldingen" />
      </Badge>
      <Dropdown items={dropdownItems} />
      <Badge className={cs.placeAd}>
        <SvgIcon className={cs.placeAdIcon} fill="white" name="pushpin" width={15} height={15} />
        <Button className={cs.placeAdButton}>Plaats Advertentie</Button>
      </Badge>
    </div>
  )
}

Menu.defaultProps = menuDefaultProps

Menu.propTypes = menuPropTypes

export default Menu
