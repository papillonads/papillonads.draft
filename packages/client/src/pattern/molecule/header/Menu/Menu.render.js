import React from 'react'
import Menu from './Menu'
import homePageData from '../../../../../../server/src/data/page/home'

const defaultRender = <Menu data={homePageData.header.navigationBar.dropdown} />

export { defaultRender }
