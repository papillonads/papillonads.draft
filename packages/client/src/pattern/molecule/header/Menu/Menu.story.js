import { storiesOf } from '@storybook/react'
import { defaultRender } from './Menu.render'
import { getInfo } from '../../../../../../../.storybook/library'

const defaultRenderInfo = getInfo('default')

storiesOf('Patttern/Molecule/Header/Menu', module).add(defaultRenderInfo.text, () => defaultRender, defaultRenderInfo.parameters)
