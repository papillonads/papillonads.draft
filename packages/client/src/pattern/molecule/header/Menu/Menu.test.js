import { defaultRender } from './Menu.render'

describe('<Menu />', () => {
  describe('Snaphot', () => {
    test('must match defaultRender', () => {
      expect(global.renderToJSON(defaultRender)).toMatchSnapshot()
    })
  })
})
