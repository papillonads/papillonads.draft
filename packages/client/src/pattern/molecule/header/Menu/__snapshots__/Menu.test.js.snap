// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Snaphot must match defaultRender 1`] = `null`;
